sdp/processors/datasets/ytdlp/create_initial_manifest.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import subprocess
+from pathlib import Path
+import json
+from sdp.logging import logger
+from sdp.processors.base_processor import BaseParallelProcessor, DataEntry
+
+
+class CreateInitialManifestytdlp(BaseParallelProcessor):
+    """
+    Processor for creating an initial dataset manifest by saving youtube links.
+    Make sure to install yt-dlp tool before funning this code. 
+
+    Tool link: https://github.com/yt-dlp/yt-dlp
+
+    Args:
+        raw_data_dir (str): Root directory of the files to be added to the manifest. Recursively searches for files with the given 'extension'.
+        output_field (str): Field to store the file paths in the dataset. Default is "audio_filepath".
+        extension (str): Extension of the files to include in the dataset. Default is "wav".
+        **kwargs: Additional keyword arguments for the base class `BaseParallelProcessor`.
+    """
+
+    def __init__(
+        self,
+        raw_data_dir: str,
+        output_field: str = "audio_filepath",
+        # extension: str = "wav",
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.raw_data_dir = Path(raw_data_dir)
+        self.output_field = output_field
+        file_path = "sdp/processors/datasets/ytdlp/search_terms.json"
```

Comment:
1) If you don't use an "extension", remove it from both the commented version and from documentation of the function
2) For convenience we use "key" instead of "field". Replace output_field with output_key
3) Maybe it would be more convenient to change file_path to more informative name? Also could you remove it's hardcoding, make it a variable passed from config file with default value?